### PR TITLE
feat(): add immediately close button functionality to the interstitial ads

### PR DESCRIPTION
**Pull request by gunhansancar**
feat(): add immediately close button functionality to the interstitial ads

### CODE REVIEW

File: app/src/androidTest/java/tv/superawesome/demoapp/interaction/CommonInteraction.kt [hunk 1]
```diff
@@ -36,6 +37,10 @@ object CommonInteraction {
         }
     }
 
+    fun launchActivityWithSuccessStub(testData: TestData, settings: () -> Unit) {
```
Comment (thomas-james-1986) [inline app/src/androidTest/java/tv/superawesome/demoapp/interaction/CommonInteraction.kt:40]: 👍

File: app/src/androidTest/java/tv/superawesome/demoapp/interaction/CommonInteraction.kt [hunk 2]
```diff
@@ -56,6 +61,10 @@ object CommonInteraction {
             .perform(click())
     }
 
+    fun clickItemAt(testData: TestData) {
+        clickPlacementById(testData.placement)
```
Comment (thomas-james-1986) [inline app/src/androidTest/java/tv/superawesome/demoapp/interaction/CommonInteraction.kt:65]: 👍 nice

File: app/src/main/java/tv/superawesome/demoapp/MainActivity2.kt [hunk 3]
```diff
@@ -64,6 +77,32 @@ class MainActivity2 : Activity() {
         configureListView()
     }
 
+    private fun updateSettings() {
+        val app = application as? MyApplication ?: return
+        val settings = app.settings
```
Comment (thomas-james-1986) [inline app/src/main/java/tv/superawesome/demoapp/MainActivity2.kt:82]: I ended up moving this to private function in my PR so we can call it when disabling playback:

```
    private fun getSettings(): SettingsData? {
        return (application as? MyApplication)?.settings
    }
```

Comment (gunhansancar) [reply app/src/main/java/tv/superawesome/demoapp/MainActivity2.kt:82]: Sure, if we merge your PR first, I can update that part while merging

**Review by thomas-james-1986 — APPROVED** [review]
👍

**Pull request merged**